metron-analytics/metron-profiler/src/main/java/org/apache/metron/profiler/bolt/ProfileBuilderBolt.java

```diff
@@ -246,6 +247,8 @@ private void flush(Tuple tickTuple) {
     // clear the execution state to prepare for the next window
     executor.clearState();
 
+    profileConfig.getTickUpdate().forEach((var, expr) -> executor.assign(var, expr, new HashMap<>()));
```

Comment:
I do not need to access the message, but rather just the internal state.  I have made a modification which does what I think it should do with some comments.  If you see issues, I'd love it if you could let me have it. :)

Taking a step back, the purpose of this is in situations where we need to do things like capture a trailing window of state.  In the instance of Median Absolute Deviation it is required because we need not just the distribution of the values, but also the distribution of the DEVIATION of the value from the median (which is the median of the values of the trailing window).  That window can be updated upon tick.  So, for instance, if we want 3 hours lookback, at hour 4, we'd have hours 1, 2 and 3.  At hour 5, we'd want hours 2, 3, and 4 and so on.  Constructing this trailing window lead me to this.  I could have captured this state internally in memory, but that had the issue that internal state for the window would be removed for the entire trailing window upon worker crash/restart.